manifests/freeipa_ldap_mappers.pp

```diff
@@ -0,0 +1,120 @@
+#
+# @summary setup FreeIPA LDAP mappers for Keycloak
+#
+# @example
+#   keycloak::freeipa_ldap_mappers { 'ipa.example.org':
+#     ldap_provider_id => '6bd0f68d-5618-a74d-bcab-089702f1bc80',
+#     realm            => 'EXAMPLE.ORG',
+#     groups_dn        => 'cn=groups,cn=accounts,dc=example,dc=org',
+#     roles_dn         => 'cn=groups,cn=accounts,dc=example,dc=org'
+#   }
+#
+# @param parent_name
+#   The name of the parent resource. Usually the title of the
+#   keycloak_ldap_provider.
+# @param realm
+#   Keycloak realm
+# @param groups_dn
+#   Groups DN
+# @param roles_dn
+#   Roles DN
+# @param ldap_provider_id
+#   Identifier for the LDAP provider to add this mapper to
+#
+define keycloak::freeipa_ldap_mappers
+(
+  String $realm,
+  String $groups_dn,
+  String $roles_dn,
+  String $parent_name = $title,
+  String $ldap_provider_id = $title,
```

Comment:
I think you'll have to make `ldap_provider_id` be `Optional[String] $ldap_provider_id = undef` and then in the manifest do something like this:

```puppet
$_ldap_provider_id = pick($ldap_provider_id, "${title}-${realm}")
```

The default is here: https://github.com/treydock/puppet-module-keycloak/blob/master/lib/puppet/type/keycloak_ldap_user_provider.rb#L33-L38